src/charm.py

```diff
@@ -142,6 +164,53 @@ def _run_sql_migration(self, set_timeout: bool) -> None:
         stdout, _ = process.wait_output()
         logger.info(f"Executing automigration: {stdout}")
 
+    def _update_config_restart_service(self, event: HookEvent) -> None:
+        if not self._container.can_connect():
+            event.defer()
+            logger.info("Cannot connect to Hydra container. Deferring the event.")
+            self.unit.status = WaitingStatus("Waiting to connect to Hydra container")
+            return
+
+        self.unit.status = MaintenanceStatus("Configuring resources")
+
+        try:
+            self._container.get_service(self._container_name)
+        except (ModelError, RuntimeError):
+            event.defer()
+            self.unit.status = WaitingStatus("Waiting for Hydra service")
+            logger.info("Hydra service is absent. Deferring the event.")
+            return
+
+        if self.database.is_resource_created():
```

Comment:
The method is inconsistent with somes `ifs` some `if nots` some `try/except` and ultimately hard to read. I think it'd look much cleaner with a consistent look:

```python
if not self._container.can_connect():
    self.unit.status = WaitingStatus("Waiting to connect to Hydra container")
    event.defer()
    return
if not self._database_relation_is_created:
   self.unit.status = BlockedStatus("Missing required relation with postgresql")
   return
if not self._database_is_available:
   self.unit.status = WaitingStatus("Waiting for database to be available")
   event.defer()
   return
if not self._hydra_service_is_running:
    self.unit.status = WaitingStatus("Waiting for Hydra service")
    event.defer()
    return
if not <add any condition>:
   self.unit.status = WaitingStatus("bla bla bla")
   event.defer()
   return
self._container.push(self._hydra_config_path, self._render_conf_file(), make_dirs=True)
self._container.start(self._container_name)
self.unit.status = ActiveStatus()
```

Reply:
Thanks for the suggestions, I updated it